src/actions/write/increaseUndernameCount.ts

```diff
@@ -62,6 +61,12 @@ export const increaseUndernameCount = async (
   if (incrementedUndernames > MAX_ALLOWED_UNDERNAMES) {
     throw new ContractError(MAX_UNDERNAME_MESSAGE);
   }
+  // This name's lease has expired and cannot be extended
+  if (endTimestamp + SECONDS_IN_GRACE_PERIOD <= currentBlockTime) {
+    throw new ContractError(
+      `This name has expired and must renewed before its undername support can be extended.`,
+    );
+  }
```

Comment:
moving this up - doesn't change error but consistent with others checking balance last